### PR TITLE
Add custom page targeting to Opt Out integration

### PR DESCRIPTION
**Pull request by zekehuntergreen**
## What does this change?
- adds a function `getConsentlessPageTargeting` which calls its consented counterpart, `getPageTargeting`, then reduces the resulting targeting object to just [the keys needed for Opt Out](https://docs.google.com/spreadsheets/d/15LReW8T5rjD6AbFyzlPrk6wSH5anfIgcdbVeh8xZDEs/edit?usp=sharing)
- adds these targeting key values to the request sent to Opt Out in `initConsentless`

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?
- Custom targeting is necessary to start serving live ads from Opt Out.

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/build-page-targeting.ts [hunk 1]
```diff
@@ -213,8 +214,73 @@ const getPageTargeting = (consentState: ConsentState): PageTargeting => {
...
+	// consentlessTargetingKeys. This causes type errors since TypeScript can't
+	// verify that the values are of the correct type.
+	// We use reduce to get around this.
+	const consentlessPageTargeting: ConsentlessPageTargeting = Object.entries(
+		consentedPageTargeting,
+	).reduce(
+		(consentlessPageTargeting, [key, value]) =>
+			isConsentlessKey(key)
+				? { ...consentlessPageTargeting, [key]: value }
+				: consentlessPageTargeting,
```
Comment (arelra) [inline static/src/javascripts/projects/common/modules/commercial/build-page-targeting.ts:274]: Minor feel free to ignore:

- do we need a unit test or is simple enough here to not need a unit test?
- the reduce spreading and creating a new object each iteration is _probably_ pretty slow and a bit hard to read. What do you think of something like:

```
Object.fromEntries(Object.entries(consentedPageTargeting).filter(({k ,v}) => isConsentlessKey(key)))
```

**Review by arelra — APPROVED** [review]

File: static/src/javascripts/projects/common/modules/commercial/build-page-targeting.ts (hunk 1 above)
Comment (zekehuntergreen) [reply static/src/javascripts/projects/common/modules/commercial/build-page-targeting.ts]: Thanks, this is much easier to read!

File: static/src/javascripts/projects/common/modules/commercial/build-page-targeting.ts [hunk 2]
```diff
@@ -213,8 +214,73 @@ const getPageTargeting = (consentState: ConsentState): PageTargeting => {
...
+	'k',
+	'refreshed',
+	'rp',
+	's',
+	'se',
+	'sens',
+	'sh',
+	'si',
+	'skinsize',
+	'slot',
```
[the remark this answers is not in the document]
Comment (zekehuntergreen) [reply static/src/javascripts/projects/common/modules/commercial/build-page-targeting.ts]: Thanks Chris - hadn't noticed that those keys aren't included in `PageTargeting`! 
Looks like `isbn` and possibly `category` might be slot-level keys as well.
I like the idea of limiting this PR to page-level targeting and following up with a separate slot-level targeting one.

**Review by chrislomaxjones — APPROVED** [review]
Looks good 👍

**Pull request merged**